pkg/controllers/user/alert/watcher/event.go

```diff
@@ -50,7 +50,7 @@ func (l *EventWatcher) Sync(key string, obj *corev1.Event) error {
 		alertID := alert.Namespace + "-" + alert.Name
 		target := alert.Spec.TargetEvent
 		if target.ResourceKind != "" {
-			if target.Type == obj.Type && target.ResourceKind == obj.InvolvedObject.Kind {
+			if obj != nil && target.Type == obj.Type && target.ResourceKind == obj.InvolvedObject.Kind {
```

Comment:
What about checking `obj!=nil` early in the function so we don't need to do the list and loop in that condition.